tripal/tripal
# TripalEntity form upgrades to match NodeForm

# Tripal 4 Core Dev Task

### Issue #2007


### Tripal Version: 4.x

## Description


This PR upgrades the TripalEntity add/edit form to match the Node add/edit form. More specifically, the node page has an "Advanced Sidebar" in both Claro and Gin admin themes that holds administrator-focused elements such as revisions, authorship, URL alias, etc. and this PR brings that sidebar to the TripalEntity form. It does this without using the TripalLayout module but also supports users adding to it using the TripalLayout module.

| TripalEntityForm | NodeForm |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/8889b617-a848-46cc-9cea-759efb387a88) | ![image](https://github.com/user-attachments/assets/193615f2-ff8b-4fd5-ad1d-c7b6655a2945) |

### Tasks

- [x] Add the Advanced group to our TripalEntityForm and populate it with metadata for the page as is done in the NodeForm
- [x] Figure out why Claro/Gin weren't theming our Advanced group the same way they did on the NodeForm and forcing them to do so 😜 
- [x] Add an "Authoring Information" tab group to the Advanced group
- [x] Remove the ability to configure the title on the TripalEntityForm in order to reduce conxusion/repetition
- [x] Add for support for the created property so that it can be included in the authorship information as it is on the node form.
- [x] Add in missing getters for the changed property so that we could display how old the content is and when it was last updated.
- [x] Update the TripalEntity::label() method to support HTML as we had done in PR #1878 since the new add/edit forms changed the titles and broke the original page title fixes. As such, I also removed the old page title fix as this solution replaces it.
- [x] Updated the base field definitions to hide some repetitive fields by default in the view

## Testing?




1. Create a new docker on this branch
2. Go to Admin > Tripal > Content > "Add Tripal Content" and then pick to add whichever one you like most. I'm choosing organism in the screenshots.
    - Confirm the page loads as expected in the screenshot
    - Fill it out and confirm it saves without error
    - Edit the page and ensure it fills in all the default without error

| Add Organism | Edit Organism |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/56eec74c-9852-45c4-950e-af3d0d591072) |  ![image](https://github.com/user-attachments/assets/e53cd867-6294-4d6c-8f17-31f0287c7271) |

3. Repeat step 2 for the other admin theme options to ensure we test Gin Light, Gin Dark, and Claro. Gin to Claro switch is done by going to "Appearance" changing the "Admin theme" drop down. Changing from Gin Light to Dark is done by going to Appearance > Settings.

| Gin Dark | Claro |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/2ac3d92a-5288-4442-85f6-8430e999f447) |  ![image](https://github.com/user-attachments/assets/141145d2-c456-48eb-a4ea-c92d2ef49299) |

4. Go to "Manage Form Display" for a Tripal Content Type and move the Author field into a field group. Confirm this pulls the author out of the sidebar.
5. Now add a new fieldgroup of type "Details Sidebar" and put some fields in it. Confirm this shows up in the advanced sidebar.

<img width="1636" alt="Screenshot 2024-11-02 at 2 43 54 PM" src="https://github.com/user-attachments/assets/1f5bd78a-7c9e-4c10-9d74-3af05dd76e55">

![image](https://github.com/user-attachments/assets/33bb8ab8-c350-4090-8be5-9e6bc8749494)

6. Now click the "Apply Tripal Layout" on the "Manage Form Display" and confirm it applies and that now when you Add/Edit content of that type, the Tripal Layout doesn't break the new changes to the form.
![image](https://github.com/user-attachments/assets/03d3d892-b4e3-4279-b0ad-d72b2a9feb98)
